fetch/fetch.go

```diff
@@ -178,68 +219,121 @@ func (f *Fetcher) FetchChainData(ctx context.Context) error {
 				// Pop the next chunk
 				f.chunkBuffer.PopFront()
 
-				wb := f.storage.WriteBatch()
+				if err := f.writeBatch(
+					item.chunk.blocks,
+					item.chunk.results,
+					item.chunkRange.from,
```

Comment:
The `from` and `to` args are used only for the logger, you can move them out of this method and log after

Reply:
I changed the method to accept a slot directly instead of it's fields separately